### PR TITLE
Mock power and cadence sensor

### PR DESCRIPTION
**Pull request by emilytrau**
## Description

Create a mock sensor (id 4) to send power and cadence values. This sensor represents the V3 DAS.js script for publishing data from the ANT+ module.

## Screenshots

```
Connected Sucessfully! Result code: 0

publishing started...

/v3/wireless-module/1/start -->  {"filename": "M1_1581910832.csv"}
started module 1

/v3/wireless-module/2/start -->  {"filename": "M2_1581910832.csv"}
started module 2

/v3/wireless-module/3/start -->  {"filename": "M3_1581910832.csv"}
started module 3

/v3/wireless-module/4/start -->  {"filename": "M4_1581910832.csv"}
started module 4

/v3/wireless-module/1/data -->  {"sensors": [{"type": "temperature", "value": 25.58}, {"type": "humidity", "value": 83.8}, {"type": "steeringAngle", "value": 10.22}]}
/v3/wireless-module/1/battery -->  {"module-id": 1, "percentage": 78.91}
/v3/wireless-module/2/data -->  {"sensors": [{"type": "co2", "value": 320.28}, {"type": "temperature", "value": 25.47}, {"type": "humidity", "value": 85.83}, {"type": "accelerometer", "value": {"x": 87.43, "y": 86.52, "z": 90.43}}, {"type": "gyroscope", "value": {"x": 86.64, "y": 86.46, "z": 94.47}}]}
/v3/wireless-module/2/battery -->  {"module-id": 2, "percentage": 82.19}
/v3/wireless-module/3/data -->  {"sensors": [{"type": "co2", "value": 326.53}, {"type": "reedVelocity", "value": 47.5}, {"type": "gps", "value": {"speed": 50.86, "satellites": 9.94, "latitude": 24.98, "longitude": 26.03, "altitude": 51.74, "course": 0.0}}]}
/v3/wireless-module/3/battery -->  {"module-id": 3, "percentage": 82.83}
/v3/wireless-module/4/data -->  {"sensors": [{"type": "power", "value": 208.46}, {"type": "cadence", "value": 76.25}]}
/v3/wireless-module/4/battery -->  {"module-id": 4, "percentage": 80.27}

/v3/wireless-module/1/data -->  {"sensors": [{"type": "temperature", "value": 24.15}, {"type": "humidity", "value": 81.15}, {"type": "steeringAngle", "value": 10.16}]}
/v3/wireless-module/2/data -->  {"sensors": [{"type": "co2", "value": 329.24}, {"type": "temperature", "value": 23.78}, {"type": "humidity", "value": 84.78}, {"type": "accelerometer", "value": {"x": 91.34, "y": 93.21, "z": 94.07}}, {"type": "gyroscope", "value": {"x": 89.11, "y": 86.93, "z": 91.56}}]}
/v3/wireless-module/3/data -->  {"sensors": [{"type": "co2", "value": 316.24}, {"type": "reedVelocity", "value": 51.42}, {"type": "gps", "value": {"speed": 48.72, "satellites": 9.55, "latitude": 25.09, "longitude": 24.59, "altitude": 48.6, "course": 0.0}}]}
/v3/wireless-module/4/data -->  {"sensors": [{"type": "power", "value": 187.6}, {"type": "cadence", "value": 80.67}]}

/v3/wireless-module/1/data -->  {"sensors": [{"type": "temperature", "value": 24.04}, {"type": "humidity", "value": 82.84}, {"type": "steeringAngle", "value": 9.54}]}
/v3/wireless-module/2/data -->  {"sensors": [{"type": "co2", "value": 311.81}, {"type": "temperature", "value": 24.78}, {"type": "humidity", "value": 86.82}, {"type": "accelerometer", "value": {"x": 89.76, "y": 90.1, "z": 93.16}}, {"type": "gyroscope", "value": {"x": 93.22, "y": 88.15, "z": 93.03}}]}
/v3/wireless-module/3/data -->  {"sensors": [{"type": "co2", "value": 314.83}, {"type": "reedVelocity", "value": 48.59}, {"type": "gps", "value": {"speed": 50.36, "satellites": 9.87, "latitude": 26.12, "longitude": 26.02, "altitude": 48.78, "course": 0.0}}]}
/v3/wireless-module/4/data -->  {"sensors": [{"type": "power", "value": 244.27}, {"type": "cadence", "value": 104.47}]}


/v3/wireless-module/1/stop -->  {}
stopped module 1

/v3/wireless-module/2/stop -->  {}
stopped module 2

/v3/wireless-module/3/stop -->  {}
stopped module 3

/v3/wireless-module/4/stop -->  {}
stopped module 4


publishing finished
```

## Steps to Test

`python mock_wireless_module.py`

### CODE REVIEW

File: Raspi/Test/mock_wireless_module.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
                    ("longitude", 25),
                    ("altitude", 50),
                    ("course", 0))
+s_power = MockSensor(200, percent_range=0.25)
```
Comment (khanguslee) [inline Raspi/Test/mock_wireless_module.py]: nit: Increase the percent range to like 80%? Would be good to be able to change how these random numbers are generated one day.

File: Raspi/Test/mock_wireless_module.py [hunk 2]
```diff
@@ -177,14 +196,15 @@ def start_modules(args):
     # TODO: Add posibility to make modules by importing a file or generating
     # random modules. The modules should not be hard coded to this script.
 
+    print()
     if args.id is None:
-        for i in range(1, 4):
+        for i in range(1, 5):
```
Comment (khanguslee) [inline Raspi/Test/mock_wireless_module.py]: Should we create a global variable to keep track of the number of modules available?

**Review by khanguslee — CHANGES_REQUESTED** [review]
Small nits. nearly done.

**Review by khanguslee — APPROVED** [review]
LGTM

**Comment by Blake-Haydon** [comment]
I have refactored the code so that this change can be done in 2 lines instead of adding a huge if statement and adding more hacks to get it to work. I think we should just add the new values to my refactored code to ensure that there are no merge errors.

File: Raspi/Test/mock_wireless_module.py [hunk 3]
```diff
@@ -44,6 +47,8 @@
                    ("longitude", 25),
                    ("altitude", 50),
                    ("course", 0))
+s_power = MockSensor(200, percent_range=0.8)
+s_cadence = MockSensor(90, percent_range=0.20)
```
Comment (Blake-Haydon) [inline Raspi/Test/mock_wireless_module.py:51]: I will add these to my PR as I've removed a lot of the hard coded elements that you have added.

Comment (Blake-Haydon) [reply Raspi/Test/mock_wireless_module.py:51]: The values are really good though

**Review by Blake-Haydon — CHANGES_REQUESTED** [review]

**Comment by emilytrau** [comment]
Implemented in #42

**Pull request closed without merging**